### ISSUE TITLE
Errors in `numba_tools` matrix functions

### ISSUE THREAD

**Issue by jllanfranchi** [IceCubeOpenSource/pisa#589]
* `conjugate`, `conjugate_transpose`:

```python
def conjugate_transpose(A, B):
    for i in range(A.shape[0]):
        for j in range(A.shape[1]):
            B[i,j] = A[j,i].conjugate()
```
should be `B[j, i] = A[i, j]` since `A` has shape `[i, j]` (as defined by loops).

* `matrix_dot_matrix`

```python
def matrix_dot_matrix(A, B, C):
    for j in range(B.shape[1]):
        for i in range(A.shape[0]):
            C[i,j] = 0.
            for n in range(C.shape[0]):
                C[i,j] += A[i,n] * B[n,j]
```
`A` has shape `[i, n]`, `B` has shape `[n, j]` and `C` will have shape `[i, j]`. Therefore, the innermost loop should iterate over `A.shape[1]` (or `B.shape[0]`).

If I'm not mistaken, these bugs luckily don't affect 3x3 matrices...

Fixes to these issues (plus unit tests for all functions and that work on both CPU and GPU) are forthcoming in https://github.com/IceCubeOpenSource/pisa/pull/585

**Comment by philippeller** [comment]
Good catch, that is indeed the wrong way. But you're right that it will not change anything in our osc. calculations as they are all 3x3

**Comment by jllanfranchi** [comment]
Fixed with pr #585 / https://github.com/IceCubeOpenSource/pisa/commit/88fabb872698c52575878df5dd7b38244d02192a
